doc/BUILDING.md

```diff
@@ -236,3 +236,7 @@ ctest --test-dir build
 
 This will build and install binary artifacts and Python sources, create
 a Python environment, install test dependencies, then execute the tests.
+
+> **Warning:**
+> We do not currently support running tests when a Python venv was used
```

Comment:
[question] Does building from a venv work fine, and will build with the venv's Python? Is it only running the Python bridge tests (i.e. embedded Python interpreter) that fails?

I wonder if switching to [`base_prefix`](https://docs.python.org/3/library/sys.html#sys.base_prefix) could be a quick fix for this:

> if site.py finds that a virtual environment is in use, the values of [prefix](https://docs.python.org/3/library/sys.html#sys.prefix) and [exec_prefix](https://docs.python.org/3/library/sys.html#sys.exec_prefix) will be changed to point to the virtual environment, whereas [base_prefix](https://docs.python.org/3/library/sys.html#sys.base_prefix) and [base_exec_prefix](https://docs.python.org/3/library/sys.html#sys.base_exec_prefix) will remain pointing to the base Python installation (the one which the virtual environment was created from).

Reply:
Yeah its just `ctest`. The build is fine. 

> I wonder if switching to [base_prefix](https://docs.python.org/3/library/sys.html#sys.base_prefix) could be a quick fix for this:

Oh good find. Thanks. I had seen that they look like the below and discounted them, but maybe that was as we already set `PYTHONHOME`? Ill try your suggestion.
```
  stdlib dir = '/Users/tomcowland/dev/OpenAssetIO/.venv311/lib/python3.11'
  sys._base_executable = '/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11'
  sys.base_prefix = '/Users/tomcowland/dev/OpenAssetIO/.venv311'
  sys.base_exec_prefix = '/Users/tomcowland/dev/OpenAssetIO/.venv311'
  sys.platlibdir = 'lib'
  sys.executable = '/Users/tomcowland/toolchain/tools/scripts/python'
  sys.prefix = '/Users/tomcowland/dev/OpenAssetIO/.venv311'
  sys.exec_prefix = '/Users/tomcowland/dev/OpenAssetIO/.venv311'
```

Reply:
Cough https://github.com/OpenAssetIO/OpenAssetIO/pull/894#discussion_r1169043615

Reply:
Fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/894/commits/a670f3299a209ae009c905c3f2a1d3c9fa99b55a).